DOC: Convert to markdown: torch.overrides.rst, type_info.rst, utils.rst, xpu.rst

loganthomas opened the pull request:
Fixes #155041 


cc @svekars @sekyondaMeta @AlannaBurke

loganthomas committed:
rename torch.overrides.rst to torch.overrides.md

loganthomas committed:
rename torch.overrides.rst to torch.overrides.md

loganthomas committed:
convert to MyST markdown format

loganthomas committed:
convert to MyST markdown format

loganthomas committed:
rename type_info.rst to type_info.md

loganthomas committed:
rename type_info.rst to type_info.md

loganthomas committed:
convert to MyST markdown format

loganthomas committed:
convert to MyST markdown format

loganthomas committed:
rename utils.rst to utils.md

loganthomas committed:
rename utils.rst to utils.md

loganthomas committed:
fix note section in type_info.md

loganthomas committed:
fix note section in type_info.md

loganthomas committed:
convert to MyST markdown format

loganthomas committed:
convert to MyST markdown format

loganthomas committed:
rename xpu.rst to xpu.md

loganthomas committed:
rename xpu.rst to xpu.md

loganthomas committed:
convert to MyST markdown format

loganthomas committed:
convert to MyST markdown format

loganthomas committed:
kick off tests

loganthomas committed:
kick off tests

zxiiro commented on docs/source/type_info.md:
```diff
@@ -1,23 +1,24 @@
-A :class:`torch.finfo` is an object that represents the numerical properties of a floating point
-:class:`torch.dtype`, (i.e. ``torch.float32``, ``torch.float64``, ``torch.float16``, and ``torch.bfloat16``). This is similar to `numpy.finfo <https://numpy.org/doc/stable/reference/generated/numpy.finfo.html>`_.
+A {class}`torch.finfo` is an object that represents the numerical properties of a floating point
+{class}`torch.dtype`, (i.e. ``torch.float32``, ``torch.float64``, ``torch.float16``, and ``torch.bfloat16``). This is similar to `numpy.finfo <https://numpy.org/doc/stable/reference/generated/numpy.finfo.html>`_.
```
I think in markdown we only need 1 backtick ` rather than 2 backticks but 2 seems to still work in any case. Do we want to update these backticks as well?

loganthomas replied:
Happy to do so!

loganthomas replied:
Actually, in this example PR that was provided for the docathon, #154438, they kept 2 backticks. I will leave as is.

AlannaBurke commented on docs/source/torch.overrides.md:
```diff
@@ -1,30 +1,49 @@
+protocol. See {ref}`extending-torch-python` for more details on the
 ``__torch_function__`` protocol.
 
-Functions
-~~~~~~~~~
-
+### Functions
```
I think updating it should be fine, as long as the headings make sense.

loganthomas committed:
Update torch.overrides.md

loganthomas committed:
Update torch.overrides.md

guangyey commented on docs/source/xpu.md:
```diff
@@ -1,8 +1,12 @@
-torch.xpu
```
Thanks for the update!

loganthomas committed:
Update type_info.md

fix tables and links

loganthomas committed:
Update type_info.md

fix tables and links

loganthomas committed:
Update type_info.md

nit strip whitespace

loganthomas committed:
Update type_info.md

nit strip whitespace

loganthomas committed:
Revert "Update type_info.md"

This reverts commit 970f363e8472818c88fe4da40f3753e259f2581f.

loganthomas committed:
Revert "Update type_info.md"

This reverts commit 970f363e8472818c88fe4da40f3753e259f2581f.

loganthomas committed:
Revert "Update type_info.md"

This reverts commit 4707dff21d6fe2134ac5182ea5b7135dc49dfca5.

loganthomas committed:
Revert "Update type_info.md"

This reverts commit 4707dff21d6fe2134ac5182ea5b7135dc49dfca5.

loganthomas committed:
fix links and tables

loganthomas committed:
fix links and tables

loganthomas committed:
Revert "fix links and tables"

This reverts commit 82eda8aa36351a77f174196a030494d3bd2e8e01.

loganthomas committed:
Revert "fix links and tables"

This reverts commit 82eda8aa36351a77f174196a030494d3bd2e8e01.

loganthomas committed:
fix tables and links

loganthomas committed:
fix tables and links

loganthomas commented:
@svekars @AlannaBurke 
I definitely used `git mv type_info.rst type_info.md` (see cc060372c8c35796541a718cf8d3b5d32f5238ef). Not sure why it's coming through as a new file and a delete 🤷‍♂️  maybe a GitHub issue? (https://github.com/orgs/community/discussions/43177)

If you click on "View File", the click on "History", there seems to be a link to the old history preserved at the bottom:
<img width="454" alt="Screenshot 2025-06-05 at 07 29 04" src="https://github.com/user-attachments/assets/7a155607-ec4e-4d89-92ae-f58136980b4d" />

- History of [`type_info.md`](https://github.com/pytorch/pytorch/commits/3ce2dd4bceceeface29ac899bed61f70dbcca34c/docs/source/type_info.md)
- History of [`type_info.rst`](https://github.com/pytorch/pytorch/commits/cc060372c8c35796541a718cf8d3b5d32f5238ef/docs/source/type_info.rst?browsing_rename_history=true&new_path=docs/source/type_info.md&original_branch=3ce2dd4bceceeface29ac899bed61f70dbcca34c)

zxiiro commented:
> @svekars @AlannaBurke I definitely used `git mv type_info.rst type_info.md` (see [cc06037](https://github.com/pytorch/pytorch/commit/cc060372c8c35796541a718cf8d3b5d32f5238ef)). Not sure why it's coming through as a new file and a delete 🤷‍♂️ maybe a GitHub issue? (https://github.com/orgs/community/discussions/43177)

Its due to how Git works. If too many lines in a file is changed Git won't find similarity between 2 files so doesn't show that they were renamed. Git doesn't actually store the history of changes in its database but rather the state of the entire workspace. I tried to explain it here hopefully it helps explain but this is expected.

https://github.com/pytorch/pytorch/pull/155120#issuecomment-2943883096

svekars reviewed (commented):
Can you please fix the tables on this page https://docs-preview.pytorch.org/pytorch/pytorch/155088/type_info.html#torch-finfo. The rest looks good.

loganthomas committed:
Merge remote-tracking branch 'upstream/main' into doc/155041-convert-to-markdown

svekars commented on docs/source/xpu.md:
```diff
@@ -42,21 +47,26 @@ Random Number Generator
-Memory management
------------------
+## Memory management
 .. autosummary::
```
```suggestion
```{eval-rst}
.. autosummary::
```

svekars committed:
Update docs/source/xpu.md

loganthomas committed:
Update xpu.md

Incorporate removal of xpu.memory found in https://github.com/pytorch/pytorch/pull/155126 changes

svekars commented:
@pytorchbot rebase

svekars commented:
@loganthomas can you please rebase to the latest main and resolve merge conflicts locally?

loganthomas committed:
Merge remote-tracking branch 'upstream/main' into doc/155041-convert-to-markdown

loganthomas committed:
Merge branch 'doc/155041-convert-to-markdown' of github.com:loganthomas/pytorch into doc/155041-convert-to-markdown

loganthomas commented:
@svekars Is there a way to preview the docs? https://docs-preview.pytorch.org/pytorch/pytorch/155088/index.html still looks stale and I can't build locally.

svekars commented:
@pytorchbot merge -f "Doc only change, doc signal green"
